lmnet/lmnet/networks/keypoint_detection/base.py

```diff
@@ -0,0 +1,205 @@
+            drawed_images[i] = visualize_keypoint_detection(drawed_images[i], joints)
+
+        return drawed_images
+
+    def _visualize_output(self, images, output, name="visualize_output"):
+        """A tensorflow mirror method for py_visualize_output().
+
+        Args:
+            images: a Tensor of shape (batch_size, height, width, 3).
+            output: a Tensor of shape (batch_size, height, width, num_joints).
+            name: str, name to display on tensorboard.
+
+        """
+
+        drawed_images = tf.py_func(self.py_visualize_output,
+                                   [images, output, self.stride],
+                                   tf.uint8)
+        tf.summary.image(name, drawed_images)
+
+    def _compute_oks(self, output, labels):
+        """Compute object keypoint similarity between output and labels.
+
+        Args:
+            output: a Tensor of shape (batch_size, height, width, num_joints).
+            labels: a Tensor of shape (batch_size, height, width, num_joints).
+
+        Returns:
+            oks: a Tensor represents object keypoint similarity.
+        """
+
+        joints_gt = self.post_process(labels)
+        joints_pred = self.post_process(output)
+
+        return tf.py_func(compute_object_keypoint_similarity,
+                          [joints_gt, joints_pred, self.image_size],
+                          tf.float32)
+
+    def summary(self, output, labels=None):
+        """Summary for tensorboard.
+
+        Args:
+            output: a Tensor of shape (batch_size, height, width, num_joints).
+            labels: a Tensor of shape (batch_size, height, width, num_joints).
+
+        """
+        images = self.images if self.data_format == 'NHWC' else tf.transpose(self.images, perm=[0, 2, 3, 1])
+        tf.summary.image("input", images)
+
+        color = np.random.randn(1, 1, 1, self.num_joints, 3)
+
+        self._colored_heatmaps(labels, color, name="labels_heatmap")
+        self._colored_heatmaps(output, color, name="output_heatmap")
+
+        self._visualize_output(images, labels, "visualize_labels")
+        self._visualize_output(images, output, "visualize_output")
+
+    def metrics(self, output, labels):
+        """Compute metrics for single-person pose estimation task.
+
+        Args:
+            output: a Tensor of shape (batch_size, height, width, num_joints).
+            labels: a Tensor of shape (batch_size, height, width, num_joints).
+
+        Returns:
+            results: a dict, {metric_name: metric_tensor}.
+            updates_op: an operation that increments the total and count variables appropriately.
+
+        """
+
+        output = output if self.data_format == 'NHWC' else tf.transpose(output, perm=[0, 2, 3, 1])
+        oks = self._compute_oks(output, labels)
+
+        results = {}
+
+        mean_oks, update_oks = tf.metrics.mean(oks)
+        updates = [update_oks]
+        updates_op = tf.group(*updates)
+        results["mean_oks"] = mean_oks
```

Comment:
Could you change this string `"mean_oks"` more self explainable? As this string will be shown in Tensorboard, it would be better to use more longer and informative name.